packages/suite/src/actions/wallet/coinjoinClientActions.ts

```diff
@@ -101,12 +102,32 @@ const onCoinjoinRoundChanged =
         // round event is triggered multiple times. like at the beginning and at the end of round process
         // critical actions should be triggered only once
         if (phaseChanged) {
+            const relatedAccountKey = coinjoinAccountsWithSession[0].key; // since all accounts share the round, any key can be used
+
             if (round.phase === RoundPhase.ConnectionConfirmation) {
-                // TODO: open critical phase modal
+                dispatch(
+                    openModal({
+                        type: 'critical-coinjoin-phase',
+                        relatedAccountKey,
```

Comment:
The modal is tracking round changes, so it would have to be dispatched with new phase every time, instead of dispatching once and then it accesses the phase itself :)

Reply:
ah, right the progress, you're aright